src/inmanta/const.py

```diff
@@ -24,6 +24,30 @@
 
 from inmanta.stable_api import stable_api
 
+# This query assumes that the resource_persistent_state table is present in the query as rps
+# It returns the status of the resource in the "status" field
+# And if it has an update in the "has_update" field
+# If a resource is deploying but has an update. It status should be "deploying" and "has_update" should be set to true.
+RESOURCE_STATUS_QUERY = """
+(
+    CASE
+        WHEN rps.is_orphan = TRUE
+            THEN 'orphaned'
+        WHEN rps.is_deploying = TRUE
+            THEN 'deploying'
+        WHEN rps.is_undefined = TRUE
+            THEN 'undefined'
+        WHEN rps.blocked = 'BLOCKED'
+            THEN 'skipped_for_undefined'
+        WHEN rps.current_intent_attribute_hash <> rps.last_deployed_attribute_hash
+            THEN 'available'
+        ELSE
+            rps.last_non_deploying_status::text
+    END
+) AS status,
+rps.current_intent_attribute_hash <> rps.last_deployed_attribute_hash AS has_update
```

Comment:
Not sure if this should be included here, on a separate const or just on a case by case basis directly in the query